record-support/src/main/com/mongodb/record/RecordTest.java

```diff
@@ -0,0 +1,209 @@
+package com.mongodb.record;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.record.entities.Address;
+import com.mongodb.record.entities.Parent;
+import com.mongodb.record.entities.Recipe;
+import com.mongodb.record.entities.School;
+import com.mongodb.record.entities.Simple;
+import com.mongodb.record.entities.Student;
+import com.mongodb.record.entities.Teacher;
+import com.mongodb.record.entities.UnderscoreId;
+import com.mongodb.record.entities.Value;
+import com.mongodb.record.entities.ValueHolder;
+import org.bson.*;
+import org.bson.codecs.BsonDocumentCodec;
+import org.bson.codecs.Codec;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.PojoCodecProvider;
+import org.bson.codecs.pojo.RecordCodecProvider;
+import org.bson.io.BasicOutputBuffer;
+import org.bson.io.ByteBufferBsonInput;
+import org.bson.io.OutputBuffer;
+import org.bson.types.ObjectId;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+public class RecordTest {
```

Comment:
This file is very ugly because it won't be merged. You can ignore the test logic, it just implements round trip testing and go straight to the tests if you want to look at them (but like I said those tests aren't comprehensive).

Reply:
Also I had to put this stuff in a new module because I couldn't get the bson module to run with Java 14 in preview mode. However, the bson module can still access the 14 preview features, if we run the code from a different module.